apps/gatsby/src/Sidebar.spec.js

```diff
@@ -26,49 +39,68 @@ const mockSdk = {
   }
-};
+});
 
 describe('Gatsby App Sidebar', () => {
+  let mockSdk;
   afterEach(cleanup);
+  beforeEach(() => {
+    mockSdk = getMockSdk();
+    mockSdk.entry.getSys.mockReturnValue(getMockContent());
+  });
 
   it('should match snapshot', () => {
     const { container } = render(<Sidebar sdk={mockSdk} />);
 
     expect(container).toMatchSnapshot();
   });
 
-  it('should debounce the fetch', async () => {
+  it('should call window.fetch and window.open with the correct urls when a user has not added a Content Sync Url', async () => {
+    const mockFetch = jest.fn(() => Promise.resolve());
+    const mockWindowOpen = jest.fn();
+    global.fetch = mockFetch;
+    global.open = mockWindowOpen;
+    const { getByText } = render(<Sidebar sdk={mockSdk} />);
+    
+    fireEvent(
+      getByText('Open Preview'),
+      new MouseEvent('click', {
+        bubbles: true,
+        cancelable: true,
+      }),
+    );
+
+    expect(mockFetch.mock.calls[0][0]).toEqual('https://webhook.com');
+    expect(mockWindowOpen.mock.calls[0][0]).toEqual('https://preview.com');
+  });
+
+  it('should call window.fetch and window.open with the correct urls when a user has added a Content Sync Url', async () => {
     const mockFetch = jest.fn(() => Promise.resolve());
-    let timerComplete = false;
+    const mockWindowOpen = jest.fn();
     global.fetch = mockFetch;
-    mockSdk.entry.onSysChanged.mockImplementationOnce(fn => {
-      fn();
-      setTimeout(() => {
-        timerComplete = true;
-      }, 1000);
-      return jest.fn();
-    });
+    global.open = mockWindowOpen;
 
-    render(<Sidebar sdk={mockSdk} />);
+    const contentSyncUrl = 'https://content-sync.com/content-sync/fake-site-id';
+    mockSdk.parameters.installation.contentSyncUrl = contentSyncUrl;
 
-    await wait(
-      () => {
-        if (timerComplete) {
-          return;
-        }
-        throw 0;
-      },
-      { timeout: 5000 }
+    const { getByText } = render(<Sidebar sdk={mockSdk} />);
+    
+    fireEvent(
+      getByText('Open Preview'),
+      new MouseEvent('click', {
+        bubbles: true,
+        cancelable: true,
+      }),
     );
 
-    expect(mockFetch).toHaveBeenCalledWith('https://webhook.com', {
-      body: '{}',
-      headers: {
-        'Content-Type': 'application/json',
-        'x-preview-auth-token': 'test-token',
-        'x-preview-update-source': 'contentful-sidebar-extension'
-      },
-      method: 'POST'
-    });
+    expect(mockFetch.mock.calls[0][0]).toEqual('https://webhook.com');
+    /**
+     * The expected url should be in the form of:
+     * {contentSyncUrl - from gatsby dashboard which includes the site id}/{the source plugin name}/{manifestId}
+     */
+    const pluginName = 'gatsby-source-contentful'
+    const expectedManifestId = '123-456-2390-08-23T15:27:27.861Z';
+    const expectedUrl = `${contentSyncUrl}/${pluginName}/${expectedManifestId}`;
+    expect(mockWindowOpen.mock.calls[0][0]).toEqual(expectedUrl);
```

Comment:
Slight nit but I think you can use toBeCalledWith https://jestjs.io/docs/expect#tohavebeencalledwitharg1-arg2-

I believe it's something like `expect(mockWindowOpen).toBeCalledWith(expectedUrl)`

Reply:
DONE